compile fails with github libzmg due to missing zmq_event_t

Trying to compile the first example in the zeromq guide gives the following error:
# gcc -lzmq hwserver.cpp -o hwserver

In file included from hwserver.cpp:6:0:
/usr/local/include/zmq.hpp:574:47: error: ‘zmq_event_t’ does not name a type
/usr/local/include/zmq.hpp:575:53: error: ‘zmq_event_t’ does not name a type
/usr/local/include/zmq.hpp:576:53: error: ‘zmq_event_t’ does not name a type
/usr/local/include/zmq.hpp:577:47: error: ‘zmq_event_t’ does not name a type
/usr/local/include/zmq.hpp:578:49: error: ‘zmq_event_t’ does not name a type
/usr/local/include/zmq.hpp:579:46: error: ‘zmq_event_t’ does not name a type
/usr/local/include/zmq.hpp:580:51: error: ‘zmq_event_t’ does not name a type
/usr/local/include/zmq.hpp:581:44: error: ‘zmq_event_t’ does not name a type
/usr/local/include/zmq.hpp:582:50: error: ‘zmq_event_t’ does not name a type
/usr/local/include/zmq.hpp:583:50: error: ‘zmq_event_t’ does not name a type
/usr/local/include/zmq.hpp:584:45: error: ‘zmq_event_t’ does not name a type
In file included from hwserver.cpp:6:0:
/usr/local/include/zmq.hpp: In member function ‘void zmq::monitor_t::monitor(zmq::socket_t&, const char*, int)’:
/usr/local/include/zmq.hpp:497:17: error: ‘zmq_event_t’ was not declared in this scope
/usr/local/include/zmq.hpp:497:29: error: expected ‘;’ before ‘msgEvent’
/usr/local/include/zmq.hpp:498:17: error: ‘msgEvent’ was not declared in this scope
/usr/local/include/zmq.hpp:500:30: error: ‘event’ was not declared in this scope

As far as I can tell, this datatype has been removed from >4.0 versions of libzmq? I can't find zmq_event_t in the header files or anywhere in the current libzmq project.
Another user had a similar issue a while ago: https://github.com/pebbe/zmq3/issues/4

The hwserver example above compiles without problems when using the czmq binding (although that's using slightly different C example code).

--- a/zmq.hpp
+++ b/zmq.hpp
@@ -62,6 +62,11 @@
 
 #if ZMQ_VERSION >= ZMQ_MAKE_VERSION(4, 1, 0)
 #define ZMQ_HAS_PROXY_STEERABLE
+/*  Socket event data  */
+typedef struct {
+    uint16_t event;  // id of the event as bitfield
+    int32_t  value ; // value is either error code, fd or reconnect interval
+} zmq_event_t;
 #endif
 
 // In order to prevent unused variable warnings when building in non-debug

